src/jitsi.ts

```diff
@@ -0,0 +1,365 @@
+
+  const inactiveTimeout = () => {
+    const participantCount = JitsiMeetJS.getNumberOfParticipants();
+
+    if (participantCount > 1) {
+      inactiveCount = 0;
+    } else {
+      inactiveCount++;
+    }
+    console.log(
+      "!!! testing inactivity: participants",
+      participantCount,
+      "and inactive count",
+      inactiveCount
+    );
+
+    if (inactiveCount >= inactiveTotal) {
+      JitsiMeetJS.executeCommand("hangup");
+    } else {
+      inactiveTimer = setTimeout(inactiveTimeout, inactiveInterval);
+    }
+  };
+  const nowActive = (event: string, params: any) => {
+    if (!inactiveInterval) {
+      return;
+    }
+    reportAction(event, params);
+    inactiveCount = 0;
+    clearTimeout(inactiveTimer);
+    inactiveTimer = setTimeout(inactiveTimeout, inactiveInterval);
+  };
+
+  if (inactiveInterval) {
+    inactiveTimer = setTimeout(inactiveTimeout, inactiveInterval);
+  }
+
+  JitsiMeetJS.on("subjectChange", (params: any) => {
+    reportAction("subjectChange", params);
+
+    if (disableBeforeUnloadHandlers) {
+      // window.addEventListener("onpagehide", (e) => { ... }) appears to be a no-op on iOS
+      // and listening for "onbeforeunload" works for both desktop and Android
+
+      if ("onbeforeunload" in window) {
+        console.log("!!! listening for beforeunload");
+        window.addEventListener("beforeunload", askOnUnload);
+      }
+    }
+
+    // (used) to reset when someone changes a media device?!?
+    if (params.subject === "") {
+      updateSubject();
+    }
+  })
+    .on("videoQualityChanged", (params: any) => {
+      reportAction("videoQualityChanged", params);
+    })
+    .on("recordingLinkAvailable", (params: any) => {
+      reportAction("recordingLinkAvailable", params);
+      recordingLink = params.link;
+
+      const ttl = Math.floor(params.ttl / 1000) || 0;
+
+      if (ttl > 0) recordingTTL = ttl;
+      updateRecTimestamp();
+    })
+    .on("recordingStatusChanged", (params: any) => {
+      reportAction("recordingStatusChanged", params);
+      if (params.on && !recordingLink) {
+        const recordings = availableRecordings();
+        const record = recordings.find((r) => r.roomName === roomName);
+
+        if (record) {
+          console.log("!!! resuming recording", record);
+          recordingLink = record.url;
+        } else {
+          console.log("!!! unable to find recording for this room");
+        }
+      }
+      updateRecTimestamp();
+      if (!params.on) {
+        recordingLink = undefined;
+      }
+    })
+    .on("readyToClose", (params: any) => {
+      reportAction("readyToClose", params);
+      window.removeEventListener("beforeunload", askOnUnload);
+      updateRecTimestamp();
+      if (inactiveTimer) {
+        clearTimeout(inactiveTimer);
+      }
+      JitsiMeetJS.dispose();
+      JitsiMeetJS = null;
+      window.open(
+        window.location.protocol + "//" + window.location.host,
+        "_self"
+      );
```

Comment:
Please add a third parameter "noopener"

Reply:
This should be OK because it's the same origin. If you don't need cross-window communication you can add the third parameter "noopener" but if you do need it it's OK to leave it.

Reply:
This is target `_self` and origin is the same so it's fine.

Reply:
I'll add `noopener` though, there's no real reason not to